enhancements/builds/build-csi-volumes.md

```diff
@@ -0,0 +1,290 @@
+I want to mount the cluster's RHEL entitlement into my builds
+so that I can `dnf install` RHEL subscritpion content in my Docker strategy build.
+
+As a developer building applications on OpenShift,
+I want to mount sealed secrets into my builds
+so that I can use credentials that are too sensitive to be exposed as a normal Kubernetes secret.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The Source and Docker strategy APIs will be extended to support `csi` as a volume source option:
+
+```yaml
+spec:
+  ...
+  dockerStrategy: # also applies to sourceStrategy
+    volumes:
+    - name: csi-ephemeral
+      mounts:
+      - destinationPath: /etc/pki/entitlement
+      source:
+        type: CSI
+        csi:
+          driver: shared-resource.csi.storage.openshift.io # from the upstream CSI volume source
+          fsType: "ext4" # optional, from upstream CSI volume source
+          nodePublishSecretRef: null # optional, from upstream CSI volume source
+          volumeAttributes: # inherited from the upstream CSI volume source
+            share: etc-pki-entitlement
+```
+
+The fields within the `csi` object are inherited from the upstream volume source
+[spec](https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/volume/).
+There is a `readOnly` field in the upstream spec which will be omitted from the build API because builds do not support writeable volumes at this time.
+
+When the build pod is created, the fields in the `csi` object will be converted to a pod CSI volume mount.
+The existin mechanisms in the Volume Mounted Resources feature will then wire the volume mount through to buildah's build environment.
+Because writeable volumes are not supported, the `readOnly` field will be set to `true` when the build pod is created.
+
+#### Tech Preview Feature Gate
+
+Clusters must opt into enabling this capability by enabling the `BuildCSIVolumes` feature gate.
+This feature gate will be added to OpenShift's tech preview set of features.
+When a cluster is installed with (or later enables) tech preview features, `csi` volumes can be added to builds.
+
+If the `BuildCSIVolumes` feature gate is not enabled, builds which use `csi` volumes should fail prior to the creation of the build pod.
+
+### Risks and Mitigations
+
+**Risk:** A vulnerability in an underlying CSI driver can let a process escape the build container.
+
+*Mitigation:*
+Upstream Kubernetes considers CSI ephemeral volumes safe, even for restricted users.
+See the [Pod Security Admission Control KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-auth/2579-psp-replacement/README.md#pod-security-standards) for justification.
+Only users with elevated permissions (admin/cluster admin) are allowed to install CSI drivers.
+The admin takes responsibility for ensuring the CSI driver is "safe".
+We will only provide a limited set of CSI drivers in the OCP payload that support CSI ephemeral volumes, which we control and vet.
```

Comment:
:+1: maybe worth calling out that setting permissions to provision CSI volumes using SCCs cannot be configured on the driver level for the reasons mentioned above.